client/js/connectionsAPI.js

```diff
@@ -0,0 +1,255 @@
+'use strict';
+/*
+ * connectionsAPI.js
+ * Copyright(c) 2016 Universidad de Las Palmas de Gran Canaria
+ * Authors:
+ *   Jaisiel Santana <jaisiel@gmail.com>
+ *   Alejandro Sánchez <alemagox@gmail.com>
+ *   Pablo Fernández <pablo.fernandez@ulpgc.es>
+ * MIT Licensed
+
+*/
+
+var Tourguide = Tourguide || {};
+
+var initConnections = function() {
+  //check if user is logged in
+  Tourguide.AJAXRequest.get('http://tourguide/client/user',
+    Tourguide.connectionsAPI.loggedIn,
+    Tourguide.connectionsAPI.notLoggedIn);
+};
+
+
+Tourguide.addLoadEvent(initConnections);
+
+ (function(app) {
+
+  var rol = {
+    endUser: 'End user',
+    restaurantViewer: 'Restaurant Viewer',
+    globalManager: 'global manager'
+  };
+
+  function loggedIn(userInfo) {
+    localStorage.setItem('userInfo', userInfo);
+
+    userInfo = JSON.parse(userInfo);
+
+    var logOutMenu = document.createElement('UL');
+    logOutMenu.classList.add('nav', 'navbar-nav', 'pull-right');
+    logOutMenu.id = 'logOutMenu';
+
+    var hiUserLi = document.createElement('LI');
+    hiUserLi.id = 'hiUser';
+    hiUserLi.className = 'menuElement';
+
+    var hiUserText = document.createElement('P');
+    hiUserText.textContent = 'Hi ' + userInfo.displayName + '!';
+
+    hiUserLi.appendChild(hiUserText);
+    logOutMenu.appendChild(hiUserLi);
+
+    var logOutLi = document.createElement('LI');
+    logOutLi.id = 'logOut';
+    logOutLi.className = 'menuElement';
+
+    var logoutLink = document.createElement('A');
+    logoutLink.id = 'logOutLink';
+    logoutLink.href = 'http://tourguide/logout';
+    logoutLink.textContent = 'Log out';
+
+    logoutLink.onclick = function() {
+      localStorage.removeItem('userInfo');
+    };
+
+    logOutLi.appendChild(logoutLink);
+    logOutMenu.appendChild(logOutLi);
+
+    document.getElementById('loggedDiv').innerHTML = '';
+
+    document.getElementById('loggedDiv').appendChild(logOutMenu);
+
+    createAndShowMenu(userInfo);
+
+    return;
+  }
+
+  function createAndShowMenu(userInfo) {
```

Comment:
I would divide this function in two, one for building the UI and another one for showing the UI, without mixing concerns